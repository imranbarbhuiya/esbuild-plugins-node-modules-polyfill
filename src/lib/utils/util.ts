export const escapeRegex = (str: string) => {
	return str.replace(/[$()*+.?[\\\]^{|}]/g, '\\$&').replace(/-/g, '\\x2d');
};

export const commonJsTemplate = ({ importPath }: { importPath: string }) => {
	return `export * from '${importPath}'`;
};

export const normalizeNodeBuiltinPath = (path: string) => {
	return path.replace(/^node:/, '').replace(/\/$/, '');
};
