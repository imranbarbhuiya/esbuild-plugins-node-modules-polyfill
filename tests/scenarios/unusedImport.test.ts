import esbuild, { type BuildOptions } from 'esbuild';

import { assertFileContent, buildAbsolutePath, createEsbuildConfig } from '../util';

import type { NodePolyfillsOptions } from '../../dist';

function createConfig(pluginOptions?: NodePolyfillsOptions): BuildOptions {
	return createEsbuildConfig(
		{
			format: 'esm',
			entryPoints: [buildAbsolutePath('./fixtures/input/unusedImport.ts')],
		},
		pluginOptions,
	);
}

describe('Unused Import Test', () => {
	test('GIVEN a file that imports from a module exporting a node builtin and does not use it THEN it should be removed by esbuild', async () => {
		const config = createConfig();

		await esbuild.build(config);

		await assertFileContent('./fixtures/output/unusedImport.js');
	});
});
