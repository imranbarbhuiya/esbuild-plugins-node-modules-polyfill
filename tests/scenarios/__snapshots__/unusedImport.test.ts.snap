// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Unused Import Test > GIVEN a file that imports from a module exporting a node builtin and does not use it THEN it should be removed by esbuild 1`] = `
"// tests/fixtures/input/unusedImport/exports.ts
var message = \\"Hello world\\";

// tests/fixtures/input/unusedImport/unusedImport.ts
console.log(message);
"
`;
